src/snark_wrapper_verifier.rs

```diff
@@ -0,0 +1,56 @@
+use std::fs;
+
+use colored::Colorize;
+use crate::{proof_from_file, VerifySnarkWrapperArgs};
+use circuit_definitions::franklin_crypto::bellman::pairing::bn256::{Bn256, Fr};
+use circuit_definitions::franklin_crypto::bellman::plonk::better_better_cs::proof::Proof;
+use circuit_definitions::{circuit_definitions::aux_layer::ZkSyncSnarkWrapperCircuit, franklin_crypto::bellman::plonk::commitments::transcript::keccak_transcript::RollingKeccakTranscript};
+
+
+
+#[derive(Clone, serde::Serialize, serde::Deserialize)]
+pub struct L1BatchProofForL1 {
+    pub aggregation_result_coords: [[u8; 32]; 4],
+    //pub scheduler_proof: Proof<Bn256, ZkSyncCircuit<Bn256, VmWitnessOracle<Bn256>>>,
+    pub scheduler_proof: Proof<Bn256, ZkSyncSnarkWrapperCircuit>,
+}
+
+pub async fn verify_snark(args: &VerifySnarkWrapperArgs) -> Result<(), String>{
+    println!("Verifying SNARK wrapped FRI proof.");
+
+    let proof: L1BatchProofForL1 = proof_from_file(&args.l1_batch_proof_file);
```

Comment:
optional nit: would be nice to also add support for fetching the proof and aux input from GCS(we already have logic that fetches those).

Reply:
yep - that is coming in the next PRs